### ISSUE TITLE
Bug: no-constant-condition false positive for while (true) in async functions containing await

### ISSUE THREAD

**Issue by Kuldeep2822k** [eslint/eslint#20905]
### Environment

Node version:  v24.15.0
npm version: 11.12.1
Local ESLint version: v10.4.0
Global ESLint version:
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

I found a false positive in ESLint's `no-constant-condition` rule.

## Minimal reproduction

**Configuration:**
```json
{
  "rules": {
    "no-constant-condition": ["error", { "checkLoops": "all" }]
  }
}
```

**Code:**
```js
async function foo() {
    while (true) {
        await bar();
    }
}
```

### What did you expect to happen?

**Expected:** No ESLint error should be reported for the `while (true)` condition.

**my reasoning**
 The loop body contains `await`, which suspends execution and allows external code to run. This is semantically identical to `YieldExpression` in generators—both allow the loop condition to potentially change between iterations. The rule already correctly handles `YieldExpression` by clearing the loop tracking set, so `AwaitExpression` should be handled identically.

### What actually happened?

**Actual:** ESLint reports:
```
Unexpected constant condition.
```

on line 2 (`while (true)`), even though the loop body contains `await` that could cause the condition to change.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiYXN5bmMgZnVuY3Rpb24gZm9vKCkge1xuICAgIHdoaWxlICh0cnVlKSB7XG4gICAgICAgIGF3YWl0IGJhcigpO1xuICAgIH1cbn1cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsibm8tY29uc3RhbnQtY29uZGl0aW9uIjpbImVycm9yIix7ImNoZWNrTG9vcHMiOiJhbGwifV19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

I think this is a clear asymmetry between two semantically equivalent control-flow suspension mechanisms (`yield` and `await`), and the fix mirrors the already-accepted pattern for `YieldExpression`.

The rule handles `YieldExpression` [here](https://github.com/eslint/eslint/blob/main/lib/rules/no-constant-condition.js#L174) but has no corresponding `AwaitExpression` handler.

**Comment by DMartens** [comment]
The intent of the rule is to find potential infinite loops.
There is an exception for `yield` as they can be used to create infinite iterables.
The difference between `yield` and `await` is that the consumer can actively control when to stop, as the generator only runs when using `generator.next()`.
